diff --git a/CMakeLists.txt b/CMakeLists.txt
index 20829bc30..5aa764308 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -725,6 +725,9 @@ else()
     set(ARCH_FLAG "-mcpu=970")
   elseif(PPC)
     set(ARCH_FLAG "-mcpu=7400")
+  elseif(IOS AND ARCH STREQUAL "x86_64")
+     message(STATUS "IOS: Changing arch from x86_64 to x86-64")
+     set(ARCH_FLAG "-march=x86-64")
   elseif(IOS AND ARCH STREQUAL "arm64")
     message(STATUS "IOS: Changing arch from arm64 to armv8")
     set(ARCH_FLAG "-march=armv8")
diff --git a/src/cryptonote_basic/cryptonote_format_utils.cpp b/src/cryptonote_basic/cryptonote_format_utils.cpp
index 946721bc7..0d3b21711 100644
--- a/src/cryptonote_basic/cryptonote_format_utils.cpp
+++ b/src/cryptonote_basic/cryptonote_format_utils.cpp
@@ -141,6 +141,21 @@ namespace cryptonote
     return h;
   }
   
+  void get_transaction_prefix_hash(const transaction_prefix& tx, crypto::hash& h)
+  {
+    std::ostringstream s;
+    binary_archive<true> a(s);
+    ::serialization::serialize(a, const_cast<transaction_prefix&>(tx));
+    crypto::cn_fast_hash(s.str().data(), s.str().size(), h);
+  }
+
+  crypto::hash get_transaction_prefix_hash(const transaction_prefix& tx)
+  {
+    crypto::hash h = crypto::null_hash;
+    get_transaction_prefix_hash(tx, h);
+    return h;
+  }
+
   bool expand_transaction_1(transaction &tx, bool base_only)
   {
     if (tx.version >= 2 && !is_coinbase(tx))
